stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -192,58 +186,83 @@ function Base.show(io::IO, ::MIME"text/plain", S::AbstractSparseMatrixCSC)
     m, n = size(S)
     print(io, m, "×", n, " ", typeof(S), " with ", xnnz, " stored ",
               xnnz == 1 ? "entry" : "entries")
-    if xnnz != 0
+    if !(m == 0 || n == 0)
         print(io, ":")
         show(IOContext(io, :typeinfo => eltype(S)), S)
     end
 end
 
 Base.show(io::IO, S::AbstractSparseMatrixCSC) = Base.show(convert(IOContext, io), S::AbstractSparseMatrixCSC)
 function Base.show(io::IOContext, S::AbstractSparseMatrixCSC)
-    nnz(S) == 0 && return show(io, MIME("text/plain"), S)
-
-    ioc = IOContext(io, :compact => true)
-    function _format_line(r, col, padr, padc)
-        print(ioc, "\n  [", rpad(rowvals(S)[r], padr), ", ", lpad(col, padc), "]  =  ")
-        if isassigned(nonzeros(S), Int(r))
-            show(ioc, nonzeros(S)[r])
+    m, n = size(S)
+    (m == 0 || n == 0) && return show(io, MIME("text/plain"), S)
+
+    # The maximal number of characters we allow to display the matrix
+    maxHeight = displaysize(io)[1] - 4 # -4 from [Prompt, header, newline after elements, new prompt]
+    maxWidth = displaysize(io)[2] ÷ 2
+
+    # In the process of generating the braille pattern to display the nonzero
+    # structure of `S`, we need to be able to scale the matrix `S` to a
+    # smaller matrix with the same aspect ratio as `S`, but fits on the
+    # available screen space. The size of that smaller matrix is stored
+    # in the variables `scaleHeight` and `scaleWidth`. If no scaling is needed,
+    # we can use the size `m × n` of `S` directly.
+    scaleHeight = m
+    scaleWidth = n
+
+    # Determine if scaling is needed and set the scaling factors
+    # `scaleHeight` and `scaleWidth` accordingly. Note that each available
+    # character can contain up to 4 braille dots in its height (⡇) and up to
+    # 2 braille dots in its width (⠉).
+    if get(io, :limit, true) && (m > 4maxHeight || n > 2maxWidth)
+        s = (m * 2maxWidth ÷ n, n * 4maxHeight ÷ m)
+        if s[1] <= 4maxHeight
+            scaleHeight = s[1]
+            scaleWidth = 2maxWidth
         else
-            print(ioc, Base.undef_ref_str)
+            scaleHeight = 4maxHeight
+            scaleWidth = s[2]
```

Comment:
Probably clearer to do:
```
 s = m * 2maxWidth ÷ n
        if s <= 4maxHeight
            scaleHeight = s
            scaleWidth = 2maxWidth
        else
            scaleHeight = 4maxHeight
            scaleWidth = n * 4maxHeight ÷ m
```

Reply:
Or just
```
s = min(2maxWidth / n, 4maxHeight / m)
scaleHeight = ceil(Int, s * m)
scaleWidth = ceil(Int, s * n)
```

Reply:
I have used `floor` instead of `ceil` to preserve the current behavior.